### ISSUE TITLE
provide "nested" groups piggy-backing on DNS

### ISSUE THREAD

**Issue by malcook** [hypothesis/product-backlog#401]
### Problem you are trying to address with this feature

 - As an H user, I might like to see all annotations by my members of my lab.  
 - On another occasion I might want to see all annotations by members of any lab at my research institute.
 - As an H group administrator, I would like to create groups that can be discovered (via a naming convention) by other members of my institute.
 - As an adherent of open annotation, I require distributed and unfettered public group administration .
 - As a organization committed to open annotation, and also seeking to "establish sustainable business models for the Hypothes.is web annotation platform", I want a "value-added" proposition to pitch to organizations once they are "sold" on H.

Imagine, where example.org is a medical research institute, a member can follow/participate:

 - journalClub.professorX.lab.example.org - professorX's journal club
 - professorX.lab.example.org - all the h.activity in her lab
 - lab.example.org - h.activity of all labs at the institute !
 - example.org - institute-wide h.activity anyone?
 - foobar.project.example.org - what are people annotating on the foobar project?

### Your solution - a new group type: "open domain groups"

 - consider groups names as extensions of domain names:   professorX.lab.example.org professorY.lab.example.org  lab1.example.org
 - require (by default) that such group names may only be created by group admin accounts having emails verified to come from the domain (i.e. you have to have a verified email from example.com to create professorX.lab.example.org).
 - allow to retrieve annotation by group, respecting the hierarchy (i.e. group="example.org" implies group %like% "example.org*")
 - develop and sell advanced controls over group admin (i.e. LDAP sync).  When/if example.org wants to exert control of group management, they will have an avenue to do so.
- develop integrations with other platforms and sell services around their configuration.   
  - a widget for [Open Science Framework](https://osf.io/)
  - Slack!  
  - Microsoft Teams!  

You could require that 
 - new top-level 'open domain" groups (named directly after the domain) only be create-able by someone coming from within the domain
 - new lower-level groups names only be create-able by h member of parent group.

This might be a nice balance of being "open" and providing an on-ramp for organizational users to "grow like topsy".  Let them admin their own internal group structure user by user, group by group, and get the whole institute "hooked" on openAnnotation, 

Then, when example.org decides they want/need to
 - make things even easier on their institutional users.  
 - tie it into other supported initiatives (library services, journal club, reference managers)
 - _control_ (shudder) this thing!

... thats when you **sell** the institutional client The WHOLE enchilada:
 - tools/configuration scripts to install the chrome extension preconfigued on all their PCs!
 - periodic group sync with auth (LDAP, AD, ...)
 - custom integration with their intranet with templates/training/evangilism for, example: "how to use H to run an efficient journal club"
 - ????

FWIW: I also am a member of F1000 which provides for sharing annotations within a group, but, alas, I can not get a group RSS feed of them though I have pitched the idea to their team.  

F1000 has a laudable literature recommendation system, but, honestly, I prefer to have an excellent open annotation system and get my literature recommendations through other means.  So, please don't use H to drive recommendation systems, or, at least, make it opt-in.  Who knows, there may be $$ for you to lash on one!

**Comment by dwhly** [comment]
I just want to drop a note of appreciation here for your suggestions.  There's a bunch of good thinking here.  We'll digest.

**Comment by malcook** [comment]
In the absence of progress on this, I'm going recommend that colleagues at my research institute collectively agree to annotate publicly and abide a convention to consistently use at least two tags:

 - TLD:stowers.org - being our top level domain
 - SCI:NNNNNN - being the ID of the research project in which we wish to surface the annotations (drawn from our internal *research project management system*)

Alas, I expect this will present hurdles which for many of my colleagues will be a barrier to adoption.  The hurdles are:

1.  the loss of control over who gets to see their idle comments, including the possibility of revealing to outsiders what they find pertinent to their research agendas
2. the overhead of adding these tags by hand to every entry
3.  it opens a vector for annotation spam (spammotation??)

Despite these hurdles, I'm going to advocate for this anyway internally at the Stowers Institute for a while and see how it goes.

With this, I can selectively surface annotations in our *research project management system* via RSS widgets.  (I am able to demo what this looks like to you if you're interested, but, they are simple RSS widgets for Atlassian Confluence and/or MS Sharepoint.)

Once exposed this way, they will be seen by all project members on a regular basis, and possibly bring some more traction to H as a platform for advancing scientific collaboration.

A workaround for 2. might be found in #799 and/or if you decide to make generally available @judell 's [custom version of the chrome plugin ](https://github.com/hypothesis/product-backlog/issues/799#issuecomment-421076452) 

A workaround for 1. might be to wait till you provide some means for controlling access based on the TLD from which a connection is coming.  I THINK THERE IS REVENUE FOR YOU HERE which might interest you given your need to [establish sustainable business models for the Hypothes\.is web annotation platform](https://web.hypothes.is/sponsors/)

Film at 11.

BTW & FWIW - I am talking to devs at F1000 (for which we have an institutional subscription) in similar terms.

**Comment by judell** [comment]
> In the absence of progress

Good news! https://github.com/hypothesis/product-backlog/issues/30 is now, finally, on the front burner:  https://github.com/hypothesis/product-backlog/projects/6

Note that although I expect API-level wildcarding will work for RSS and Atom queries, such queries are unauthenticated and will not work for non-public layers. So if you want to leverage the simplicity and generality of RSS/Atom, which sounds like a good idea to me, then your notion of recommending annotation in the public layer might still make sense. 

Alternatively, as I think you may already have tried, it's possible to provide your own RSS feeds (at open but unpublished URLs) based on authenticated Hypothesis API queries. That will enable you to leverage both key benefits of H groups: for privacy, and for scoping a queryable set of annotations (because a group id functions as a kind of automatically-assigned tag). 

I /think/ what we are planning will support sitewide query for all of your examples:

> journalClub.professorX.lab.example.org - professorX's journal club
> professorX.lab.example.org - all the h.activity in her lab
> lab.example.org - h.activity of all labs at the institute !
> example.org - institute-wide h.activity anyone?
> foobar.project.example.org - what are people annotating on the foobar project?

But let's check with @hmstepanek who is doing the implementation!

**Comment by malcook** [comment]
Yes - I've been watching #30 with interest and expecting that you would settle on stemming searches using % as wildcard as provided by your postgresql backing-store.  

I have provided my own RSS feeds thanks to your examples.  I do like that this gives me the ability to reformat them as I see fit, especially since I want each RSS post to aggregate annotations since last (hourly?) crawl on a url basis.  However, I currently can't serve them outside our firewall, and alas our use of MS Office 365 as platform requires such - I'm sure there are workarounds (involving tunneling/ports etc) but I don't own the keys to those castles and want fewer moving parts.

So!  I still think "there is money to be made" by up-selling a service to institutions (think pharma) that would freely share annotations to all clients who come from the same TLD.  Or range of IP address.  Or what have you.  And make if free to verified non-profits (like me!).  The model is dead simple and I think will come to be quite valuable - once you get enough brains annotating the web!  

But then, RSS is practically dead and I'm holding on for dear life.  WHy should I expect open annotation to really catch on?

If it were up to me, we would bring back nntp and usenet and give every doi a thread.

Feh

**Comment by malcook** [comment]
Is it likely that you will implement "domain based groups" where membership is implicitly defined based on request coming from behind registered domain?

I'm sure there are issues with details (such as, if I leave my company, do I lose access to content I've written whilst a member of a "domain based group"?

Perhaps all y'all already have a considered opinion on this matter.... ???  

Tx

**Comment by judell** [comment]
> % as wildcard as provided by your postgresql backing-store

Actually the API talks to elasticsearch, and its wildcard is '*'

> RSS is practically dead

Tragically. 

> Is it likely that you will implement "domain based groups"

Yes, this exists now. At https://europepmc.org/abstract/MED/10029393, for example, you'll see annotations in two groups: Public, and Europe PMC. The latter is a *restricted group" -- world-readable, writeable only by members, and visible (in the Hypothesis client) only on europepmc.org.

**Comment by malcook** [comment]
I think we mean different things by "domain based groups".  

My suggestion might be reduced to the following:

  - An H user can create a special "domain based group" (DBG) whose name ends in a TLD (.com, .org,. etc) only if their H account has an associated email validated to come from a domain whose name is that of the proposed group.
 - The creation of a DBG auto-notifies all H member with emails in that domain with one-click opt-in.
 - The association of an email with an H user similarly auto-notifies the user if a corresponding DBG exists.

Then you only have to allow for multiple emails to be attached to an H account (with one perhaps being "primary").

This would allow _anyone_ having an email at, say, stowers.org, (such as me) to create a group named "stowers.org" and for all H members at Stowers to easily opt in.  

FWIW and IMO - It is not your job to figure out whether I am the CIO or a librarian.  Just let me create the group.  "stowers.org" is just a string.

You need more traction / eyeballs.  This might help get some!
